src/replica/Application.cc

```diff
@@ -138,20 +135,19 @@ int Application::run() {
             " System's setups in case of an accidental mis-configuration.",
             _instanceId
         );
-    }
-    if (_injectXrootdOptions) {
-        parser().option(
-            "xrootd-allow-reconnect",
-            "Change the default XROOTD connection handling node. Set 0 to disable"
-            " automatic reconnects. Any other number would allow reconnects.",
-            _xrootdAllowReconnect
-        ).option(
-            "xrootd-reconnect-timeout",
-            "Change the default value limiting a duration of time for making automatic"
-            " reconnects to the XROOTD servers before failing and reporting error"
-            " (if the server is not up, or if it's not reachable for some reason)",
-            _xrootdConnectTimeoutSec
-        );
+        // Inject options for th egeneral configuration parameters.
+        for (auto&& itr: ConfigurationSchema::parameters()) {
+            string const& category = itr.first;
+            for (auto&& param: itr.second) {
+                // The read-only parameters can't be updated programmatically.
+                if (ConfigurationSchema::readOnly(category, param)) continue;
+                _general[category][param] = ConfigurationSchema::defaultValueAsString(category, param);
+                parser().option(
+                    category + "." + param,
```

Comment:
So instead of `--xrootd-allow-reconnect` you now have `--xrootd.allow-reconnect`? That looks a bit odd, can you keep dash separator?

Reply:
I will fix that.